cpp/oneapi/dal/algo/dbscan/backend/gpu/compute_kernel_brute_force_dpc.cpp

```diff
@@ -0,0 +1,115 @@
+/*******************************************************************************
+* Copyright 2021 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#include "oneapi/dal/algo/dbscan/backend/gpu/compute_kernel.hpp"
+#include "oneapi/dal/backend/interop/common_dpc.hpp"
+#include "oneapi/dal/backend/interop/error_converter.hpp"
+#include "oneapi/dal/backend/interop/table_conversion.hpp"
+#include "oneapi/dal/exceptions.hpp"
+#include "oneapi/dal/table/row_accessor.hpp"
+
+#include <daal/src/algorithms/dbscan/oneapi/dbscan_kernel_ucapi.h>
+
+namespace oneapi::dal::dbscan::backend {
+
+using std::int64_t;
+using dal::backend::context_gpu;
+using descriptor_t = detail::descriptor_base<task::clustering>;
+
+namespace daal_dbscan = daal::algorithms::dbscan;
+namespace interop = dal::backend::interop;
+
+template <typename Float>
+using daal_dbscan_t = daal_dbscan::internal::DBSCANBatchKernelUCAPI<Float>;
+
+template <typename Float, typename Task>
+static compute_result<Task> call_daal_kernel(const context_gpu& ctx,
+                                             const descriptor_t& desc,
+                                             const table& data,
+                                             const table& weights) {
+    auto& queue = ctx.get_queue();
+    interop::execution_context_guard guard(queue);
+
+    const int64_t row_count = data.get_row_count();
+    const int64_t column_count = data.get_column_count();
+
+    const double epsilon = desc.get_epsilon();
+    const int64_t min_observations = desc.get_min_observations();
+
+    daal_dbscan::Parameter par(epsilon, dal::detail::integral_cast<std::size_t>(min_observations));
+    par.memorySavingMode = desc.get_mem_save_mode();
+    par.resultsToCompute = daal_dbscan::computeCoreIndices | daal_dbscan::computeCoreObservations;
+
+    const auto daal_data = interop::convert_to_daal_table<Float>(data);
+    const auto daal_weights = interop::convert_to_daal_table<Float>(weights);
+
+    array<int> arr_responses = array<int>::empty(row_count * 1);
+    array<int> arr_cluster_count = array<int>::empty(1);
```

Comment:
Shouldn't we allocate device memory?

Reply:
Added allocation of shared USM: device one caused problem in DAAL kernel (inside context.fill)

Reply:
Done